src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourcesPage.cs

```diff
@@ -0,0 +1,243 @@
+        {
+            var packageSourcesList = value as IList<IDictionary<string, object>>;
+            if (packageSourcesList is null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            try
+            {
+                // Stop listening to setting changes while saving.
+                _suppressSettingValuesChanged = true;
+
+                switch (moniker)
+                {
+                    case MonikerPackageSources:
+                        return await Task.Run(
+                            () => SavePackageSources<T>(packageSourcesList, cancellationToken),
+                            cancellationToken);
+
+                    case MonikerMachineWideSources:
+                        return await Task.Run(
+                            () => SetIsEnabledOnMachineWidePackageSources(packageSourcesList, cancellationToken),
+                            cancellationToken);
+
+                    default:
+                        throw new InvalidOperationException();
+                }
+            }
+            finally
+            {
+                // Resume listening to setting changes after saving.
+                _suppressSettingValuesChanged = false;
+            }
+        }
+
+        private ExternalSettingOperationResult SetIsEnabledOnMachineWidePackageSources(
+            IList<IDictionary<string, object>> packageSourcesList,
+            CancellationToken cancellationToken)
+        {
+            ExternalSettingOperationResult result;
+
+            try
+            {
+                var machineWidePackageSources = LoadPackageSources(isMachineWide: true);
+
+                foreach (PackageSource originalMachineWideSource in machineWidePackageSources)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    string originalPackageSourceName = originalMachineWideSource.Name;
+                    IDictionary<string, object> targetPackageSource = packageSourcesList
+                        .Single(packageSourceDictionary =>
+                            packageSourceDictionary[MonikerSourceName].ToString() == originalPackageSourceName);
+
+                    bool originalIsEnabled = originalMachineWideSource.IsEnabled;
+                    bool targetIsEnabled = (bool)targetPackageSource[MonikerIsEnabled];
+
+                    if (originalIsEnabled != targetIsEnabled)
+                    {
+                        if (targetIsEnabled)
+                        {
+                            _packageSourceProvider.EnablePackageSource(originalPackageSourceName);
+                        }
+                        else
+                        {
+                            _packageSourceProvider.DisablePackageSource(originalPackageSourceName);
+                        }
+                    }
+
+                    // Only one value can be set at a time from Unified Settings, so we're done once a change is made.
```

Comment:
It looks like this will only check the first element of machineWidePackageSources and then stop, regardless of whether that element was changed.  Am I missing something?

Reply:
That's right, because there's only 1 edit possible at a time (eg, Checked/Unchecked the Is Enabled checkbox). 
Once a difference is found, there's no reason to continue the loop, because there's no Bulk Edit capability in Unified Settings, today.